spec/models/asp/payment_request_validator_spec.rb

```diff
@@ -154,4 +148,39 @@
 
     include_examples "invalidation", :needs_abrogated_attributive_decision
   end
+
+  context "when the student is missing biological sex" do
+    before { asp_payment_request.student.update!(biological_sex: nil) }
+
+    include_examples "invalidation", :missing_biological_sex
+  end
+
+  context "when the student is missing birthplace city INSEE code and is born in France" do
+    before do
```

Comment:
tu peux faire un contexte en plus 

```
when the student is born in France
  let(:student) { create(:student, :born_in_france) } # pas besoin d'encoder le `99100`
  and the city is missing
  ```

Reply:
Tu confonds 'birth_country_code' et 'birthplace_country_insee_code'

Reply:
je confonds rien du tout : si ton test a besoin d'un élève né en France il faudrait qu'il utilise `create(:student, :born_in_france)`

Reply:
Il y a une subtilité ici c'est que tout le fichier utilise un student born in france mais ici on check 'birthplace_country_insee_code' ce qui est différent, je te jure. 

Par contre il manquait un test sur "missing_birthplace_country_insee_code" que je viens de rajouter.

Reply:
OK bah j'ai juste pas compris mais en fait le point que je défend c'est qu'on devrait jamais utiliser `99100` direct dans le code, si t'as besoin d'un élève né en France on laisse à la factory la responsabilité de savoir ce que ça représente (i.e `build(:student, :born_in_france)`) comme on lui laisse la responsabilité de nous fournir un élève qui ne l'est pas (`build(:student, :born_abroad)`) ça permet de DRY et d'éviter de semer cette logique dans la test suite.